aopy/preproc/wrappers.py

```diff
@@ -346,18 +378,123 @@ def proc_broadband(data_dir, files, result_dir, result_filename, overwrite=False
         # Append the broadband metadata to the file
         aodata.save_hdf(result_dir, result_filename, metadata, "/broadband_metadata", append=True)
 
-def proc_lfp(data_dir, files, result_dir, result_filename, overwrite=False, max_memory_gb=1., filter_kwargs={}):
+def proc_spikes(data_dir, files, result_dir, result_filename, kilosort_dir=None, overwrite=False):
+    '''
+    Process spike data:
+        neuropixels: synchronize spike times with ecube. Extract waveforms from drift corrected data. 
+
+    Saves spike times and waveforms into the HDF datasets:
+        spikes (HDF5 group (dict)) : spike times for each drive
+        waveforms (HDF5 group (dict)) : spike times for each drive
+        metadata (dict)
+    
+    Args:
+        data_dir (str): where the data files are located
+        files (dict): dictionary of filenames indexed by system
+        result_dir (str): where to store the processed result
+        result_filename (str): where to store the processed result
+        kilosort_dir (str): data folder where preprocessed data of kilosort is saved
+        overwrite (bool, optional): whether to remove existing processed files if they exist
+        filter_kwargs (dict, optional): keyword arguments to pass to :func:`aopy.precondition.filter_lfp`
+    '''  
+    # Check if a processed file already exists
+    filepath = os.path.join(result_dir, result_filename)
+    if not overwrite and os.path.exists(filepath):
+        contents = aodata.get_hdf_dictionary(result_dir, result_filename)
+        if "spikes" in contents:
+            print("File {} already preprocessed, doing nothing.".format(result_filename))
+            return
+    elif os.path.exists(filepath):
+        os.remove(filepath)
+
+    # Check if record_headstage is True or False
+    record_headstage_key = 'record_headstage'
+    metadata_group = 'exp_metadata'
+    tmp = result_filename.split('_')[:-1]
+    exp_filename = "_".join(tmp) + '_exp.hdf'
+    record_headstage = aodata.load_hdf_data(result_dir, exp_filename, record_headstage_key, metadata_group, cached=True)
+
+    idrive = 0
+    # Preprocess neural data into lfp   
+    if 'neuropixels' in files and not record_headstage:
```

Comment:
why do you care if record_headstage is true or false? if neuropixels is present then you should process it i think

Reply:
also is there a way to check if kilosort has been run? because having 'neuropixels' in files only tells you that neuropixels data was recorded but you don't know that it's been sorted